documentation/linux-setup.md

```diff
@@ -0,0 +1,221 @@
+# Linux environment setup
+
+Sometimes it may be preferable to perform local environment maintenance directly using tooling installed on your Linux OS (or in [WSL](https://learn.microsoft.com/en-us/windows/wsl/about)) instead of the maintenance docker container. This guide documents how to set up an environment in that way. It also includes ways you can confirm that every step was completed successfully.
+
+## Operating system
+
+This guide is written for [Ubuntu 22.04](https://releases.ubuntu.com/jammy/). Since that is the operating system used in the maintenance container, Github runners, and Azure runners it is a good choice of OS for this. This guide will also work when using [Ubuntu in WSL](https://canonical-ubuntu-wsl.readthedocs-hosted.com/en/latest/guides/install-ubuntu-wsl2/). Other Ubuntu variants or Linux distributions may work as well but are not documented here.
+
+Double check:
+
+```
+lsb_release -a
+```
+
+## Clone the repository
+
+[Git](https://git-scm.com/) should have been installed with your operating system automatically. Clone the repository from GitHub and change into the directory.
+
+```
+git clone https://github.com/GCTC-NTGC/gc-digital-talent.git
+cd gc-digital-talent
+```
+
+Double-check:
+
+```
+ls -a
+```
+
+You should see the same list of files visible in [the github repository online](https://github.com/GCTC-NTGC/gc-digital-talent).
+
+## Docker compose
+
+[Docker](https://www.docker.com/) compose is used to run the services for this app. It should have been installed with your operating system automatically.
+
+If you haven't before, add your user to the `docker` group and activate the group:
+
+```
+sudo usermod -aG docker $USER
+newgrp docker
+```
+
+Double-check:
+
+```
+groups
+```
+
+## Hosts file
+
+We need to ensure that scripts running locally can find the services running in Docker.
+
+```
+sudo nano /etc/hosts
```

Comment:
Nano?! :exploding_head:

Reply:
Sadly, Neovim isn't in the default install.  :laughing: